Dockerfile

```diff
@@ -0,0 +1,69 @@
+# ===============================================
+FROM helsinkitest/node:12-slim as appbase
+# ===============================================
+# Offical image has npm log verbosity as info. More info - https://github.com/nodejs/docker-node#verbosity
+ENV NPM_CONFIG_LOGLEVEL warn
+
+# set our node environment, either development or production
+# defaults to production, compose overrides this to development on build and run
+ARG NODE_ENV=production
+ENV NODE_ENV $NODE_ENV
+
+# Global npm deps in a non-root user directory
+ENV NPM_CONFIG_PREFIX=/app/.npm-global
+ENV PATH=$PATH:/app/.npm-global/bin
+
+# Yarn
+ENV YARN_VERSION 1.19.1
+RUN yarn policies set-version $YARN_VERSION
+
+# Use non-root user
+USER appuser
+
+# Copy package.json and package-lock.json/yarn.lock files
+COPY package*.json *yarn* ./
+
+# Install npm dependencies
+ENV PATH /app/node_modules/.bin:$PATH
+
+USER root
+
+RUN apt-install.sh build-essential
+
+USER appuser
+RUN yarn && yarn cache clean --force
+
+USER root
+RUN apt-cleanup.sh build-essential
```

Comment:
Removing stuff in a separate layer does not make the image smaller. There two possible solutions:

1. Do everything in a single layer
```Dockerfile
USER root

RUN apt-install.sh build-essential \
    && su -c "yarn && yarn clean cache --force" appuser \
    && apt-cleanup.sh build-essential
```
2. Use a separate stage
```Dockerfile
FROM helsinkitest/node:12-slim as appbuilder

RUN apt-install.sh build-essential
COPY package.json yarn.lock ./
RUN yarn

FROM helsinkitest/node:12-slim as appbase

COPY --from=appbuilder --chown=appuser:appuser <SRC> <DST>
```

Reply:
Ok thanks, I'll go with the first one.

Reply:
Made the change. Impact was pretty nice, size went from 895MB to 736MB.